### PR TITLE
Refresh the ./dtox.sh dev env.

### PR DESCRIPTION
**Pull request by jsirois**
This ensures the dev env is useable for running tests, inspecting test
failures and iterating. It also fills out the Python interpreter set to
cover all interpreters Pex supports.

### CODE REVIEW

**Comment by jsirois** [comment]
I'll be using the base image for Linux CI here shortly to alleviate a GH action cache size limit by caching in an image layer from a cronned image re-build on top of the base image or as a seperate data volume image. The long-standing `./dtox.sh ...<tox args>...` works well including supporting the new `--devpi` option. The new `./dtox.sh inspect` inspect mode and `TMPDIR` trick for preserving pytest temp dirs works well.

File: dtox.sh [hunk 1]
```diff
@@ -7,40 +7,59 @@ ROOT="$(git rev-parse --show-toplevel)"
...
-  --tty \
-  --rm \
-  --volume $(pwd):/dev/pex \
-  pantsbuild/pex:user \
-  "$@"
+    --interactive \
+    --tty \
+    --rm \
+    --volume "${HOME}/.netrc:/home/$(id -un)/.netrc" \
+    --volume "${HOME}/.ssh:/home/$(id -un)/.ssh" \
```
Comment (jsirois) [inline dtox.sh:53]: These are needed for various ITs involving netrc support and GH repo cloning (VCS requirement tests).

File: dtox.sh [hunk 2]
```diff
@@ -7,40 +7,59 @@ ROOT="$(git rev-parse --show-toplevel)"
 BASE_INPUT=(
   "${ROOT}/docker/base/Dockerfile"
 )
-base_hash=$(cat ${BASE_INPUT[@]} | git hash-object -t blob --stdin)
+base_hash=$(cat "${BASE_INPUT[@]}" | git hash-object -t blob --stdin)
```
Comment (jsirois) [inline dtox.sh:10]: If you haven't seen this script before, the hashing / labeling / short-circuiting are all to make `./dtox.sh` startup latency not too bad - at least on Linux:
```
$ hyperfine 'tox --version' './dtox.sh --version'
Benchmark 1: tox --version
  Time (mean ± σ):     105.1 ms ±   2.4 ms    [User: 94.7 ms, System: 10.2 ms]
  Range (min … max):   101.2 ms … 110.2 ms    28 runs

Benchmark 2: ./dtox.sh --version
  Time (mean ± σ):      1.223 s ±  0.041 s    [User: 0.025 s, System: 0.089 s]
  Range (min … max):    1.160 s …  1.284 s    10 runs

Summary
  'tox --version' ran
   11.64 ± 0.47 times faster than './dtox.sh --version'
```

**Comment by jsirois** [comment]
Sorry about that if you were observing before the force-push. Pushed some tail commits I intended for another branch!

**Review by kaos — APPROVED** [review]
lgtm :)

**Pull request merged**